internal/biz/resources/resources.go

```diff
@@ -214,11 +214,10 @@ func updateExistingReporterResource(ctx context.Context, m *model.Resource, exis
 func (uc *Usecase) Check(ctx context.Context, permission, namespace string, sub *kessel.SubjectReference, id model.ReporterResourceId) (bool, error) {
 	res, err := uc.reporterResourceRepository.FindByReporterResourceId(ctx, id)
 	if err != nil {
-		if errors.Is(err, gorm.ErrRecordNotFound) {
-			// resource doesn't exist.
-			return false, nil
+		if !errors.Is(err, gorm.ErrRecordNotFound) {
```

Comment:
I think it would be good to leave some generous comments in the code as to what we're doing here, since the intent is not obvious and the changes should probably be reversed in the future when the data is consistent. It would also help to capture the intent for anyone who ends up refactoring this code in the short term.